frontend/lib/src/components/shared/StreamlitMarkdown/styled-components.ts

```diff
@@ -43,27 +43,31 @@ function sharedMarkdownStyle(theme: Theme): any {
 
 /**
  * Caption sizes taken from default styles, but using em instead of rem, so it
- * inherits the <small>'s shrunk size
- *
+ * inherits the <small>'s shrunk size. Also handles reduced heading font sizes
+ * in dialogs.
  */
 function convertFontSizes(
   fontSize: string,
-  smallFontSize: string,
-  useSmallerHeadings: boolean,
+  isInDialog: boolean,
   isCaption: boolean
 ): string {
-  if (useSmallerHeadings) {
-    // For headers in `st.caption`, we use `em` values, so the headers automatically
-    // become a bit smaller by adapting to the font size of the caption.
-    return isCaption ? convertRemToEm(smallFontSize) : smallFontSize
+  // For headers in `st.caption`, we use `em` values, so the headers automatically
+  // become a bit smaller by adapting to the font size of the caption.
+
+  if (isInDialog) {
+    // Dialogs also reduce the font size of the headings
+    const roundedFontSize = Math.round(parseFloat(fontSize) * 0.65 * 8) / 8
```

Comment:
nit: since the rem rounding to the nearest 8th is also used below, maybe put this into a small utils function with an explanatory comment on why we do this.